### PR TITLE
GovernanceSlasher test Foundry Migration

### LINKED ISSUE #10813: test/governance/network/governance_slasher.ts

### PR DESCRIPTION
**Pull request by soloseng**
### Description

migrated the governance slasher test to foundry.

### Tested

Unit tested

### Related issues

- Fixes #10813

### CODE REVIEW

File: packages/protocol/test-sol/governance/network/GovernanceSlasher.t.sol [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+import "../../../contracts/common/Accounts.sol";
+import "../../../contracts/common/FixidityLib.sol";
+import "../../../contracts/common/interfaces/IRegistry.sol";
+import "../../../contracts/governance/Proposals.sol";
+import "../../../contracts/governance/test/MockLockedGold.sol";
+import "../../../contracts/governance/test/MockValidators.sol";
+
+// Contract to test
+import "../../../contracts/governance/GovernanceSlasher.sol";
```
Comment (martinvol) [inline packages/protocol/test-sol/governance/network/GovernanceSlasher.t.sol]:
```suggestion
import "@celo-contracts/contracts/common/Accounts.sol";
import "@celo-contracts/contracts/common/FixidityLib.sol";
import "@celo-contracts/contracts/common/interfaces/IRegistry.sol";
import "@celo-contracts/contracts/governance/Proposals.sol";
import "@celo-contracts/contracts/governance/test/MockLockedGold.sol";
import "@celo-contracts/contracts/governance/test/MockValidators.sol";

// Contract to test
import "@celo-contracts/contracts/governance/GovernanceSlasher.sol";
```

File: packages/protocol/test-sol/governance/network/GovernanceSlasher.t.sol [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+  function setUp() public {
+    owner = address(this);
+    nonOwner = actor("nonOwner");
+    validator = actor("validator");
+    slashedAddress = actor("slashedAddress");
+
+    accounts = new Accounts(true);
+    mockLockedGold = new MockLockedGold();
+    mockValidators = new MockValidators();
```
Comment (martinvol) [inline packages/protocol/test-sol/governance/network/GovernanceSlasher.t.sol]: Are you sure you need the mocked version? I couldn't find a reference but maybe is in one calls by other contracts.

File: packages/protocol/test-sol/governance/network/GovernanceSlasher.t.sol [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    governanceSlasher = new GovernanceSlasher(true);
+
+    deployCodeTo("Registry.sol", abi.encode(false), registryAddress);
```
Comment (martinvol) [inline packages/protocol/test-sol/governance/network/GovernanceSlasher.t.sol:40]: Do you need to force deployment to this address?

**Review by martinvol — APPROVED** [review]

File: packages/protocol/test-sol/governance/network/GovernanceSlasher.t.sol (hunk 3 above)
Comment (soloseng) [reply packages/protocol/test-sol/governance/network/GovernanceSlasher.t.sol:40]: I dont need to.

However, to maintain consistency across our tests, I thinks its a good idea to set to the registry address to the one we expect.

File: packages/protocol/test-sol/governance/network/GovernanceSlasher.t.sol (hunk 2 above)
Comment (soloseng) [reply packages/protocol/test-sol/governance/network/GovernanceSlasher.t.sol]: I was keeping it similar to the original test.

Technically, setting the validator address in the registry is not required for the set of test we are running. 

Removing it completly from this test

**Pull request merged**